.github/workflows/ci.yml

```diff
@@ -0,0 +1,110 @@
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+          token: ${{ secrets.ERADIATECI_PAT }}
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniconda-version: latest
+          activate-environment: eradiate
+      - name: Get Mitsuba hash
+        run: |
+          cd ext/mitsuba
+          echo "MITSUBA_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+      - name: Cache Mitsuba
+        uses: actions/cache@v2
+        with:
+          path: build
+          key:
+            mitsuba-${{ env.MITSUBA_HASH }}-${{hashFiles('src/plugins/src/*/*.cpp', 'src/plugins/src/*/*.h')}}-${{ env.CACHE_NUMBER}}
+        env:
+          CACHE_NUMBER: 0
+        id: cache
+      - name: Build
+        shell: bash -l {0}
+        run: |          
+          sudo apt-get update && DEBIAN_FRONTEND=noninteractive sudo apt-get install -y   \
+          git              \
+          ninja-build      \
+          clang-11          \
+          libc++-11-dev     \
+          libc++abi-11-dev  \
+          libpng-dev       \
+          zlib1g-dev       \
+          build-essential  \
+          libjpeg-dev      \
+          ninja-build
+
+          export CUDA_VISIBLE_DEVICES=""
+          conda install mamba -n base -c conda-forge
+          make mamba-init
+          conda deactivate
+          conda activate eradiate
+
+          conda install cmake=3.22
+          cmake --version
+          
+          export CC=clang-11
+          export CXX=clang++-11
+
+          cmake --preset default
+          cmake --build build
+          python -c "import drjit; import mitsuba; mitsuba.set_variant('scalar_rgb')"
+        if: steps.cache.outputs.cache-hit != 'true'
+  test:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    needs: build
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+          token: ${{ secrets.ERADIATECI_PAT }}
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniconda-version: latest
+          activate-environment: eradiate
+      - name: Get Mitsuba hash
+        run: |
+          cd ext/mitsuba
+          echo "MITSUBA_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+      - name: Cache Mitsuba
+        uses: actions/cache@v2
+        with:
+          path: build
+          key:
+            mitsuba-${{ env.MITSUBA_HASH }}-${{hashFiles('src/plugins/src/*/*.cpp', 'src/plugins/src/*/*.h')}}-${{ env.CACHE_NUMBER}}
+        env:
+          CACHE_NUMBER: 0
+      - name: run tests
+        shell: bash -l {0}
+        run: |
+          export CUDA_VISIBLE_DEVICES=""
+          conda install mamba -n base -c conda-forge
```

Comment:
How about using the Mamba setup [suggested by the `setup-miniconda` action docs](https://github.com/marketplace/actions/setup-miniconda#example-6-mamba)?

Reply:
Edit: Actually, I just noticed that there is also an example for lock file usage: https://github.com/marketplace/actions/setup-miniconda#example-7-lockfiles

Maybe this is even better?